repair/row_level.cc

```diff
@@ -1050,7 +1083,12 @@ class repair_meta {
     future<std::tuple<std::list<repair_row>, size_t>>
     read_rows_from_disk(size_t cur_size) {
         using value_type = std::tuple<std::list<repair_row>, size_t>;
-        return do_with(cur_size, size_t(0), std::list<repair_row>(), [this] (size_t& cur_size, size_t& new_rows_size, std::list<repair_row>& cur_rows) {
+        if (!_reader_permit) {
+            bool all_shards = !_is_local_reader;
+            auto permit = co_await get_global_reader_permit(_rs, this_shard_id(), all_shards);
+            _reader_permit = std::optional<repair_reader_permit_meta>(std::move(permit));
+        }
```

Comment:
I think this permit should be taken when creating the repair meta. We save a lot more memory that way.
With this mechanism we effectively limit the concurrency of repair-metas.

Reply:
I thought about this too. Will the readers consume a lot of memory even before we read from it for the first time? If this is not the case, it is not going to safe much memory.

Reply:
Not really. The multishard reader in particular creates its shard readers on-demand. If some shard is never reached, it won't create a shard reader for it at all.

Reply:
Good. Then I think it is good to take the permit when we actually read something.'
For multiple reasons: 
-  I do not want to block the "short" verbs which creates repair instance on the follower nodes
- We need to store the "permits" back to the repair_meta even if we create it outside
- We need new logic outside repair_meta to know if the repair_meta is going to create local or mutlti-shard reader.